Fixes #22592 - Add component for radio buttons

xprazak2 opened the pull request:
Radio buttons when used in form:

![](https://i.imgur.com/jFqzgju.png)

Commit:
Fixes #22592 - Add component for radio buttons

xprazak2 commented on webpack/assets/javascripts/react_app/components/common/forms/RadioButtonGroup.js:
```diff
@@ -0,0 +1,24 @@
+RadioButtonGroup.propTypes = {
+  controlLabel: PropTypes.string.isRequired,
+  radios: PropTypes.arrayOf(PropTypes.shape({ value: PropTypes.string })),
+  name: PropTypes.string
```
Srsly?

xprazak2 commented:
Fixing linting errors.

lizagilman commented:
Shouldn't we base on patternfly-react components?
[patternfly-react form controls](https://rawgit.com/patternfly/patternfly-react/gh-pages/index.html?knob-Show%20Modal=true&knob-Close%20Popover=true&knob-Show%20Labels=false&knob-Show%20Help=true&knob-Stacked=false&knob-Show%20Loading=false&knob-Disabled=false&knob-Field%20Level%20Help%20Content=Please%20specify%20Country%20code%20%3Cbr%3E%20%3Ca%20target%3D%27_blank%27%20href%3D%27https%3A%2F%2Fcountrycode.org%2F%27%3EClick%20here%20for%20a%20list%20of%20Country%20codes%3C%2Fa%3E&knob-fluid=true&knob-ShowClearfix=true&selectedKind=Forms&selectedStory=Supported%20Controls&full=0&addons=1&stories=1&panelRight=0&addonPanel=storybooks%2Fstorybook-addon-knobs )
@sharvit

sharvit commented:
Thanks @xprazak2 
I feel confused as @lizagilman, `patternfly-react` have all the relevant components for rendering a `RadioButtonGroup` and it makes sense to use them.

It is also make sense to wrap the `patternfly-react` components with a data-driver appraoch :+1: 

Can you please add a storybook so it will be easier to review?

dLobatog reviewed (dismissed):
Set to WoC, @xprazak2 I also wonder why do the Patternfly-react radio buttons not fit your use case?

boaz0 commented on webpack/assets/javascripts/react_app/components/common/forms/RadioButtonGroup.js:
```diff
@@ -0,0 +1,27 @@
+
+RadioButtonGroup.propTypes = {
+  controlLabel: PropTypes.string.isRequired,
+  radios: PropTypes.arrayOf(PropTypes.shape({ value: PropTypes.string })),
```
I am not sure but maybe it is a good idea to mention here that each radio has `label`?
WDYT?

xprazak2 commented:
Updated, using `Radio` from patternfly-react and the look is exactly the same:
![](https://i.imgur.com/BgYrBki.png)

What is the benefit of using patternfy-react in this particular case? I fail to see it, because now I have to depend on a component from external library and it did not shorten my code.

I also added label to propTypes.

sharvit commented:
@xprazak2 I think there are some reasons

1. It is easier to consume components with there props instead consuming css classes
2. You used to be depended on patternfly before (using their css) but without an explict connection (harder to fallow where your design came from)
3. Most important, the Radio component will be shared between projects and will be meintained more often from that reason

Since this component never used, can you please wrap it in a storybook? It would be a nice way to demonstrate it.

xprazak2 commented:
I added a storybook.

xprazak2 commented:
~~Do not merge, investigating a possible bug~~

Nope, works as expected.

xprazak2 commented:
Added a prop that supports disabling the whole group, minor change in storybook to pass code climate.

sharvit commented on webpack/stories/storeDecorator.js:
```diff
@@ -0,0 +1,7 @@
+
+const storeDecorator = getStory => (<Provider store={Store}>{getStory()}</Provider>);
+
+export default storeDecorator;
```
Nice :+1:

sharvit commented on webpack/assets/javascripts/react_app/components/common/forms/RadioButtonGroup.js:
```diff
@@ -0,0 +1,30 @@
+
+RadioButtonGroup.propTypes = {
+  controlLabel: PropTypes.string.isRequired,
+  radios: PropTypes.arrayOf(PropTypes.shape({ value: PropTypes.string, label: PropTypes.string })),
```
Can you add the `checked` to the shape please?

xprazak2 commented:
Added `checked` into shape.

xprazak2 commented:
@dLobatog, any additional comments?

amirfefer commented on webpack/assets/javascripts/react_app/components/common/forms/RadioButtonGroup.test.js:
```diff
@@ -0,0 +1,21 @@
+  it('should render group of radio buttons', () => {
+    const wrapper = shallow(<RadioButtonGroup name="RadioButtonGroupTest" controlLabel="RadioButtonGroupLabel" radios={radios} />);
+    expect(toJson(wrapper)).toMatchSnapshot();
+  });
```
could you use `Prettier` format ?
for example this test should look like:
```javascript
    const wrapper = shallow(
      <RadioButtonGroup
        name="RadioButtonGroupTest"
        controlLabel="RadioButtonGroupLabel"
        radios={radios}
      />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
```

xprazak2 replied:
Running Prettier out of the box gives me incompatibilities with our linting rules. Formatting the code in test manually according to your example gives me still linting offences, namely:

![](https://i.imgur.com/QLTkYSI.png)

I am very confused. Did we decide to use Prettier as a style guide? If so, then why don't we have a [pinned version](https://prettier.io/docs/en/install.html)  in `package.json` and CI job that would check and enforce the style?

If not, then why require ad-hoc code styling? Don't we have linting for this?

amirfefer replied:
I know that there were some attemptes to combine lint and Prettier with no success.
the best practice for now is to run Prettier first and lint after.

sharvit replied:
`npm run lint -- --fix` usually fix the errors from prettier

xprazak2 commented:
I ran this through Prettier and `eslint --fix`

mmoll commented:
@xprazak2 please rebase :)

xprazak2 commented:
Rebased

sharvit reviewed (approved):
LGTM!

ohadlevy commented on webpack/assets/javascripts/react_app/components/common/forms/Form.stories.js:
```diff
@@ -0,0 +1,38 @@
+
+  return (
+    <Form>
+      <RadioButtonGroup name="steak" controlLabel="Would you like a steak?" radios={radios}/>
```
can we change it to: would you like an hamburger? https://www.youtube.com/watch?v=lLuc6rtWkrM

xprazak2 replied:
Changed steak to hamburger :)

ohadlevy commented:
@dLobatog ack?

xprazak2 commented:
Added missing key prop for iterator children.

tstrachota reviewed (approved):
I tested the component and it works nicely. The code looks good to me too.

@dLobatog do you have any further comments?

tstrachota commented:
Merging, thank you @xprazak2 !
